docs/src/rules/no-restricted-globals.md

```diff
@@ -106,3 +106,25 @@ function onClick() {
 ```
 
 :::
+
+Restricted globals used in TypeScript type annotations—such as type references, interface inheritance, or class implementations—are ignored by this rule.
+
+Examples of **correct** TypeScript code for sample "Promise", "Event", and "Window" global variable names:
+
+::: correct
+
+```ts
+/*eslint no-restricted-globals: ["error", "Promise", "Event", "Window"]*/
+
+const fetchData: Promise<string> = window.Promise.resolve("data");
```

Comment:
I think having Promise on both sides of the assignment here makes this a bit confusing (ideally we'd also capture `window.Promise`, but alas).

```suggestion
const fetchData: Promise<string> = fetchString();
```

Reply:
FYI checking implicit globals when referenced on their explicit global would break a use my team has for the rule (to enforce consistency for how some implicit globals are referenced).

I know this was likely just a passing remark, but wanted to mention in case such a change is considered at some point in the future - I feel like it'd need to be an option!